SofaKernel/modules/SofaBaseMechanics/src/SofaBaseMechanics/DiagonalMass.h

```diff
@@ -308,7 +308,7 @@ class DiagonalMass : public core::behavior::Mass<DataTypes>
     SReal getElementMass(sofa::Index index) const override;
     void getElementMass(sofa::Index, defaulttype::BaseMatrix *m) const override;
 
-    bool isDiagonal() override {return true;}
+    bool isDiagonal() const override {return true;}
```

Comment:
so in fact siDiagonal return true only for DiagonalMass...  at least this is consistent!